src/Security/LogOutAuthenticationHandler.php

```diff
@@ -58,16 +58,10 @@ public function logOut(HTTPRequest $request = null)
         $loginHandler = Injector::inst()->get(LogInAuthenticationHandler::class);
         $member = Security::getCurrentUser();
 
-        if (RememberLoginHash::config()->get('logout_across_devices')) {
-            foreach ($member->LoginSessions() as $session) {
-                $session->delete();
-            }
-        } else {
-            $loginSessionID = $request->getSession()->get($loginHandler->getSessionVariable());
-            $loginSession = LoginSession::get()->byID($loginSessionID);
-            if ($loginSession && $loginSession->canDelete($member)) {
-                $loginSession->delete();
-            }
+        $loginSessionID = $request->getSession()->get($loginHandler->getSessionVariable());
+        $loginSession = LoginSession::get()->byID($loginSessionID);
+        if ($loginSession && $loginSession->canDelete($member)) {
+            $loginSession->delete();
```

Comment:
Just clarifying my understanding of this here. 

Previously, if `logout_across_devices` is `true` and you pressed to logout button, it would kill all your sessions, which would immediately log you out of all sessions irrespective of whether you had check the "remember me" checkbox .

Now,  if `logout_across_devices` is `true` and you pressed to logout button, it will not immediately log you out of existing sessions. It will however invalidate all your remember me tokens.

Did I get this right? Is that the intended behaviour?

Reply:
Both points are correct

The 'logout immediately' bit is handled by session manager, and only targets a single device

The 'remember me token' bit is handled within framework only